routes/users.js

```diff
@@ -0,0 +1,94 @@
+const express = require('express');
+const usersRouter = express.Router();
+const db = require('../database');
+
+usersRouter.get('/', (req, res) => {
+    res.send('Users');
+});
+usersRouter.get('/:id', (req, res) => {
+    db.raw(`SELECT * FROM users WHERE id = ?;`, [req.params.id])
+        .then((results) => {
+            if (results.rows.length === 0) {
+                res.send(400, 'Not found')
+            } else {
+                res.json(results.rows);
+            }
+        })
+        .catch((err) => {
+            res.send('This is an error.')
+        });
+});
+usersRouter.post('/login', (req, res) => {
+    let email = req.body.email;
+    let pw = req.body.password;
+    db.raw("SELECT * FROM users WHERE email = ?;", [email])
+        .then((results) => {
+            if (results.rows.length === 0) {
+                return res.send({ "message": "User not found." })
+            } else if (results.rows[0].password === pw) {
+                return res.json(results.rows);
+            } else {
+                return res.send({ "message": "Incorrect password." });
+            };
+        })
+        .catch((err) => {
+            console.log(err);
+            res.status(500);
+        });
+});
+usersRouter.post('/register', (req, res) => {
+    let email = req.body.email;
+    let pw = req.body.password;
+    db.raw("SELECT * FROM users WHERE email = ?", [email])
+        .then((results) => {
+            if (results.rows.length !== 0) {
+                return res.send({ "message": "User already exists." });
+            }
+            db.raw(`INSERT INTO users (email, password) VALUES (?, ?) RETURNING *;`, [email, pw])
```

Comment:
Return your db promises, then the output of the promise will be in the .then of the outer promise. This helps from nesting your promises.